### PR TITLE
feat(core): Main event getter module

### PR DESCRIPTION
**Pull request by kartikCypherock**
Fixes https://app.clickup.com/t/9002019994/CHI-4747

### CODE REVIEW

File: common/core/events.c [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  bool p0_evt_occurred = false;
+  bool p1_evt_occurred = false;
+
+  /* Poll for the selected events, until atleast one event is captured. */
+  while (1) {
+    p0_evt_occurred = p0_get_evt(&(p_evt_status->p0_event));
+
+    if (evt_config.evt_selection.bits.ui_events) {
+      lv_task_handler();
+      BSP_DelayMs(50);
```
[the remark this answers is not in the document]
Comment (kartikCypherock) [reply common/core/events.c:112]: If the screen does not generate any event.. will it ever call this function `get_events`?

**Comment by rohanagarwal94** [comment]
Task linked: [CHI-4747 Implement OS Event getter](https://app.clickup.com/t/9002019994/CHI-4747)

File: common/core/events.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (ujjwal-cyph) [reply common/core/events.c:112]: Okay, lets take this up when we really face a need for such behavior requirement.

**Comment by kartikCypherock** [comment]
I am approving this PR with following assumption: UI event API will provide an interface to discard any UI events

**Pull request merged**